Add X509_alias_set1 to the ffi

ajoino opened the pull request:
Add `X509_alias_set1` to the ffi (not sure about the wording) as a first step to allow serialization of CA friendlyNames in the PKCS12 format. See Issue #6135, in particular [this post](https://github.com/pyca/cryptography/issues/6135#issuecomment-868449674) for more information.

ajoino commented:
This is my first time trying to contribute to this project, and I am unsure what kinds of tests I need to write for this change. I tried looking around but couldn't find any tests of the similar `X509_alias_get0` function.

reaperhulk commented:
We don't write tests for just bindings; compilation against our many configurations is sufficient to know that the symbol is available.

The current CI failure is due to #6898. Once that lands you can rebase this and we can merge it.

ajoino commented:
Thanks @reaperhulk, I will ask again for help when it comes to the rebasing, I haven't actually done that in any of my own projects :grimacing:.

One more question, in #6135 there were some ideas regarding a pkcs12 builder. I want to try and work on that next, do you think I should discuss it in the mailing list first or just submit a PR and discuss it there?

reaperhulk commented:
You don't have to actually rebase, you can simply merge main into your branch once it updates (but if you want to rebase you can check out main again, pull the latest, check out your branch, and type `git rebase main` and it will do it).

Feel free to submit a PR, but you can also propose an API before implementing it in the issue (or file a new issue to discuss if there isn't an appropriate issue in place). There has been some commentary about what a good API might look like in the past so I'd be interested in seeing how your approach matches with that.

reaperhulk commented:
(This can be merged/rebased now)

ajoino commented:
I think I did things correctly, please tell me if something went wrong.

reaperhulk commented:
Looks like the rebase/merge somehow pulled in the new commits but GH is showing a diff including those commits (which it would not normally). Could you do a `git pull upstream main` with your branch checked out? Where `upstream` is an alternate remote. You will need to `git remote add upstream https://github.com/pyca/cryptography` to add that upstream. That should merge main and then you can push that here. It will likely be a weird pile of commits but we squash anyway 😄

ajoino commented:
So the issue might be that when I rebased I had to create a new commit, which seemed weird to me at the time but now I know that is wrong. I think I fixed the issue by resetting and creating a new commit with the same changes. Hopefully I got it this time :smile:

reaperhulk commented:
(CI is misbehaving so going to just close/reopen to force some triggers)
